openshiftcontrolplane/v1/types.go

```diff
@@ -117,6 +117,21 @@ type RoutingConfig struct {
 	Subdomain string `json:"subdomain"`
 }
 
+// ImportModeType describes how to import an image manifest.
+type ImportModeType string
```

Comment:
This needs the enum tags too

Reply:
i can add it - but i was unsure because these fields are set by apiserver based on the different crd statuses and not controlled directly.

Reply:
We should have the validation even if they are set by controllers